File: admin-portal/models/events/event.js
const mongoose = require("mongoose");

const schema = new mongoose.Schema({
  event_title: {
    type: String,
    required: true,
  },
  event_description: {
    type: String,
    required: true,
  },
  event_date: {
    type: String,
    required: true,
  },
  event_date_posted: {
    type: String,
    required: true,
  },
  picture_url: {
    type: String,
  },
  picture_public_id: {
    type: String,
  },
});

module.exports = mongoose.model("event", schema);

File: green_cleats/lib/apps/PlayersApp/More/Events/Event.dart
class Event {
  String? eventId;
  String? event_title;
  String? event_description;
  String? event_date;
  String? event_date_posted;
  String? pictureURL;

  Event({
    this.eventId,
    this.event_title,
    this.event_description,
    this.event_date,
    this.event_date_posted,
    this.pictureURL,
  });

  factory Event.fromJson(Map<String, dynamic> json) {
    return Event(
      eventId: json['_id'],
      event_title: json['event_title'],
      event_description: json['event_description'],
      event_date: json['event_date'],
      event_date_posted: json['event_date_posted'],
      pictureURL: json['picture_url'],
    );
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['_id'] = eventId;
    data['event_title'] = event_title;
    data['event_description'] = event_description;
    data['event_date'] = event_date;
    data['event_date_posted'] = event_date_posted;
    data['picture_url'] = pictureURL;
    return data;
  }
}
